### DISCUSSION TITLE
exec on startup permissions

### DISCUSSION THREAD

**Discussion by Bastinator-school** [i3/i3#6170] [Q&A] [answered]
first of all i know that i can exec things on startup via i3 config thats what i do now but i have a script that i need to run as my user. because it needs to make changes inside my home directory. Is this possible through the i3 config?

**Comment by orestisfl** [comment]
i3 runs commands as the same user as it is started, so it should work out of the box. Do you not run i3 using your user?

**Reply by Bastinator-school** [reply]
yes i do my issue is when i run my script via cmd path it works as expected but when run via i3 it messes up i thought it might be because of permissions

**Reply by orestisfl** [reply]
if you share your config & script, I can take a look

**Reply by Bastinator-school** [reply]
sure give me a sec i have to cesor some stuff thx

**Reply by Bastinator-school** [reply]
```bash
#!/bin/bash

LAST_RUN_FILE="/home/$USER/.last_login_run_date"
SCRIPT_TO_RUN="/home/$USER/.configupdate.sh"
TODAY=$(date +%Y-%m-%d)

if [ ! -f "$LAST_RUN_FILE" ]; then
    echo "$TODAY" > "$LAST_RUN_FILE"
    bash "$SCRIPT_TO_RUN"
else
    LAST_RUN_DATE=$(cat "$LAST_RUN_FILE")
    if [ "$TODAY" != "$LAST_RUN_DATE" ]; then
        echo "$TODAY" > "$LAST_RUN_FILE"
        bash "$SCRIPT_TO_RUN"
    fi
fi
```

```bash
#!/bin/bash

sudo rm /home/$USER/.Workstation-debian.git.old
sudo mv /home/$USER/.Workstation-debian.git /home/$USER/.Workstation-debian.git.old

git clone https://token@github.com/Bastinator-school/Workstation-debian.git /home/$USER/.Workstation-debian.git

cp /home/$USER/.Workstation-debian/config /home/$USER/.config/i3/config
sudo cp /home/$USER/.Workstation-debian/i3status.conf /etc/i3status.conf

touch /home/$USER/.last_login_run_date
```

**Reply by Bastinator-school** [reply]
the first script is to be executed by i3 and intern execs the 2nd i know that it does'nt get to touch /home/$USER/.last_login_run_date

**Reply by orestisfl** [reply, marked as answer]
The problem is with the second script, you are calling `sudo` which requires authentication that you can't provide when i3 runs it in the background.

Files under $HOME should be owned by your user so there is no need to `sudo rm` or `sudo mv`

For this command:
```
sudo cp /home/$USER/.Workstation-debian/i3status.conf /etc/i3status.conf
```

I would recommend running i3status using a config file under $HOME: `~/.config/i3status/config`

**Reply by Bastinator-school** [reply]
thx for the help i changed config location and it worked after some tweaking
